src/diff_heat_map.ts

```diff
@@ -135,105 +135,119 @@ class DiffHeatMap implements IAppView {
-        .data(d.structure.added_rows)
-        .enter()
-        .append('div')
-        .attr('class', 'taco-added-row')
-        .attr('class', 'struct-add-color')
-        .attr('title', function (d) {
-          return d.id;
-        })
-        .style('left', 0 + 'px')
-        .style('top', function (d) {
-          const y = d.pos;
-          return (y !== -1 ? y * h : null) + 'px';
-        })
-        .style('width', width + 'px')
-        .style('height', h + 'px');
-
-      const addedCols = root.selectAll('.taco-added-col')
-        .data(d.structure.added_cols)
-        .enter()
-        .append('div')
-        .attr('title', function (d) {
-          return d.id;
-        })
-        .attr('class', 'taco-added-col')
-        .attr('class', 'struct-add-color')
-        .style('top', 0 + 'px')
-        .style('left', function (d) {
-          const x = d.pos;
-          return (x !== -1 ? x * w : null) + 'px';
-        })
-        .style('width', w + 'px')
-        .style('height', height + 'px');
-
-      const deletedRows = root.selectAll('.taco-del-row')
-        .data(d.structure.deleted_rows)
-        .enter()
-        .append('div')
-        .attr('class', 'taco-del-row')
-        .attr('class', 'struct-del-color')
-        .attr('title', function (d) {
-          return d.id;
-        })
-        .style('left', 0 + 'px')
-        .style('top', function (d) {
-          const y = d.pos;
-          return (y !== -1 ? y * h : null) + 'px';
-        })
-        .style('width', width + 'px')
-        .style('height', h + 'px');
-
-
-      const deletedCols = root.selectAll('.taco-del-col')
-        .data(d.structure.deleted_cols)
-        .enter()
-        .append('div')
-        .attr('class', 'taco-del-col')
-        .attr('class', 'struct-del-color')
-        .attr('title', function (d) {
-          return d.id;
-        })
-        .style('top', 0 + 'px')
-        .style('left', function (d) {
-          const x = d.pos;
-          return (x !== -1 ? x * w : null) + 'px';
-        })
-        .style('width', w + 'px')
-        .style('height', height + 'px');
-
-      const chCells = root.selectAll('.taco-ch-cell').data(d.content);
-      chCells.enter()
-        .append('div')
-        .attr('class', 'taco-ch-cell')
-        .attr('title', function (d) {
-          return '(' + d.row + ',' + d.col + ': ' + d.diff_data + ')';
-        })
-        .style('top', function (d) {
-          //var y = that.row_ids.indexOf(d.row);
-          const y = d.rpos;
-          return (y !== -1 ? y * h : null) + 'px';
-        })
-        .style('left', function (d) {
-          //var x = that.col_ids.indexOf(d.col);
-          const x = d.cpos;
-          return (x !== -1 ? x * w : null) + 'px';
-        })
-        .style('width', w * 10 + 'px')
-        .style('height', h * 10 + 'px')
-        .style('background-color', 'red');
-      /* .style('background-color', function (d) {
+      if (d.hasOwnProperty('structure')) {
```

Comment:
You can simply write `if (d.structure) {`. That will check if `structure` is defined and not `null` on `d`.
That's the same for all other occurences.